src/sorted_set_test.cc

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
```

Comment:
Why are you saying that Google owns the copyright for code that Google did not write? The copyright is yours, unless you substantially derived this file from another copyrighted work.

Attributing copyright in this way without a proper contributor license agreement is bad form, and can cause a lot of issues if someone were to ever attempt to enforce the terms of the Apache license for this copyrighted code.

Reply:
Should I write then `Copyright 2019 Konstantin Kharlamov. All Rights Reserved`?

Reply:
I am not a lawyer, so you're going to have to make that determination for yourself.

But, my non-lawyer advise, is unless someone is paying you to write software (e.g., a work for hire), you are the one who owns the copyright for it.

Reply:
> In this case, this entire file appears to be something you wrote yourself. So unless you copied it from someone / somewhere else, my non-lawyer expectation is that you would be the owner of the copyright for it.

Thanks. I was thinking that contributing a code to a project makes the project to own the code.

Reply:
My understanding of international copyright law is, obviously, not perfect, but I've discussed it over the years with various lawyers during college, and also in the business world, so I feel reasonably confident in saying that copyright can potentially be handled on a per-line-of-code basis.

The default behavior of copyright is:

Copyright is automatic, and inherent to the creation of a work. No registration needed. The mere act of creating something grants you the copyright for it.
Copyright is not transferred automatically unless there is an explicit agreement otherwise. This might be because someone is paying you (work for hire) or a contributor license agreement where you've explicitly agreed that anything you submit to a project via a pull request is automatically also a full transfer of copyright, so that it becomes their property and not yours.

In this case, you are the owner of the copyright for this file. And if anyone ever breached the license for this copyrighted code (the Apache 2.0 license, specifically), you have the right to personally take them to court for violating that license.

It's important for open source projects to know who the copyright holders are, because "A project" can't sue violators of the license. Only the owners of the copyright can.

It's also important because some licenses don't allow for certain types of re-licensing without explicit permission. I'm not fully up to speed on the Apache2.0 license's limitations on re-licensing, but I know it's very tricky with GPLv2 licensed projects, involving someone needing to track down every, single, contributor, ever, and get their written approval, OR the code in question would need to be removed from the project in order to re-license it.

Reply:
Are you a lawyer?

Reply:
He says he isn't, but he talked with them a lot.

Reply:
GitHub only showed me the first comment somehow ... nvm.